main.cpp

```diff
@@ -134,7 +150,149 @@ send me a DM to check your pull request
  Wait for my code review.
 */
 
-#include <iostream>
+
+// =============================================================
+//                  Numeric object definition
+// =============================================================
+template <typename TemplatedType> 
+struct Numeric
+{
+public:    
+    using CurrentType = Temporary<TemplatedType>;
+    
+    // Constructor when no argument is passed
+    Numeric() 
+    : value( std::make_unique<CurrentType>(0) ) {}
+
+    // Constructor when passing an object
+    Numeric(CurrentType value_) 
+    : value( std::make_unique<CurrentType>(value_) ) {}
+    
+    // -------------------------------
+    // Operators overloading
+    // -------------------------------
+    // Assign operator
+    template <typename OtherTemplatedType>
+    Numeric& operator=(const OtherTemplatedType& ott)
+    {
+        *value = static_cast<TemplatedType>(ott);
+        return *this;
+    }    
+    // Add-equal operator
+    template <typename OtherTemplatedType>
+    Numeric& operator+=(const OtherTemplatedType& ott)
+    {
+        *value += static_cast<TemplatedType>(ott);
+        return *this;
+    }
+    // Subtract-equal operator
+    template <typename OtherTemplatedType>
+    Numeric& operator-=(const OtherTemplatedType& ott)
+    {
+        *value -= static_cast<TemplatedType>(ott);
+        return *this;
+    }
+    // Multiply-equal operator
+    template <typename OtherTemplatedType>
+    Numeric& operator*=(const OtherTemplatedType& ott)
+    {
+        *value *= static_cast<TemplatedType>(ott);
+        return *this;
+    }
+    // Divide-equal operator
+    template <typename OtherTemplatedType>   
+    Numeric& operator /= (const OtherTemplatedType& ott)
+    {
+        if constexpr ( std::is_same<int, OtherTemplatedType>::value ) 
+        {
+            if( ott == 0 )
+            {
+                std::cout << "Division by zero not allowed with integers!!" << std::endl;
+                return *this;
+            }
+        }
+        else if( std::abs(ott) <= std::numeric_limits<OtherTemplatedType>::epsilon() )
+        {
+            std::cout << "Warning: Floating point division by zero!" << std::endl;
+        }
+
+        *value /= static_cast<TemplatedType>(ott);
+        return *this;
```

Comment:
The instructions for this function were ultimately revised to this: 
```
#11) now that your class is templated, you'll need to adjust your logic in your division function to handle if your input is a zero or not, based on your templated type.  
        - look up how to use std::is_same<>::value on cppreference to determine the type of your template parameter.
        
        - look up how to use std::numeric_limits<>::epsilon() to determine if you're dividing by a floating point 0
        
        - read about Knuth's algorithm here: https://www.learncpp.com/cpp-tutorial/relational-operators-and-floating-point-comparisons/

        - in plain-english, you'll need to implement logic like this:
        if your template type is an int
                if your parameter's type is also an int
                        if your parameter is 0
                                don't do the division
                else if it's less than epsilon
                        dont do the divison
        else if it's less than epsilon
                warn about doing the division

        - to make these checks work during compilation, your if() statements will need to be 'constexpr':  if constexpr (expression)
```
- [ ] revise according to final instructions.